m4/systemd.m4

```diff
@@ -128,3 +128,66 @@ AC_DEFUN([AX_AVAILABLE_SYSTEMD], [
 	AX_CHECK_SYSTEMD_DETECT_AND_ENABLE()
 	AX_CHECK_SYSTEMD()
 ])
+
+AC_DEFUN([AX_CHECK_SYSTEMD_FEATURES], [
+        if test x"$systemd" = "xy"; then
```

Comment:
perhaps we should `AC_MSG_ERR` if this macro is called when `systemd` is not `y`? This makes it mandatory to call one of the `AX_SYSTEMD_*` macros beforehand (as we can't use `AC_REQUIRE`).

Reply:
I'm wondering if we should't move that to a separate m4 file, I have searched a while for something similar, and I believe it might be of use in other projects.